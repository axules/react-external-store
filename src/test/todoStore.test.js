import { TodoStoreClass } from './TodoStore';
import { testDelay } from './utils';


describe('todoStore', () => {
  let todoStore = null;

  beforeEach(() => {
    todoStore = new TodoStoreClass({
      items: [
        { id: 't1', text: 'Todo item 1', done: false },
        { id: 't2', text: 'Todo item 2', done: true },
        { id: 't3', text: 'Todo item 3', done: true },
        { id: 't4', text: 'Todo item 4', done: false },
      ],
      status: { loading: false, fail: false }
    });
  });

  test('snapshot', () => {
    expect(todoStore).toMatchSnapshot();
  });

  test('should trigger change trigger 2 times', async () => {
    const listener = jest.fn();
    todoStore.__subscribe__(listener);
    await todoStore.loadItems();

    await testDelay(10);
    expect(listener).toBeCalledTimes(2);
  });
});