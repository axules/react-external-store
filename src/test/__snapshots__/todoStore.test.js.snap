// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoStore snapshot 1`] = `
TodoStoreClass {
  "__emitChangesTask__": [Function],
  "__emitChangesTrigger__": null,
  "__emitChanges__": [Function],
  "__listeners__": [],
  "__logger": [Function],
  "__state__": {
    "items": [
      {
        "done": false,
        "id": "t1",
        "text": "Todo item 1",
      },
      {
        "done": true,
        "id": "t2",
        "text": "Todo item 2",
      },
      {
        "done": true,
        "id": "t3",
        "text": "Todo item 3",
      },
      {
        "done": false,
        "id": "t4",
        "text": "Todo item 4",
      },
    ],
    "status": {
      "fail": false,
      "loading": false,
    },
  },
  "__subscribe__": [Function],
  "__unsubscribe__": [Function],
  "beforeUpdate": undefined,
  "getState": [Function],
  "mergeState": [Function],
  "patchState": [Function],
  "setState": [Function],
  "use": [Function],
  "useMemoized": [Function],
}
`;
